packages/react-components/src/components/chart/hooks/useVisualizedDataStreams.ts

```diff
@@ -22,24 +23,41 @@ export const useVisualizedDataStreams = (queries: TimeSeriesDataQuery[], passedI
     },
   });
 
-  // Line | Scatter | Bar charts do not support alarm streams.
-  const dataStreamsWithoutAlarms = dataStreams.filter(isNotAlarmStream);
-
-  const hasError = useMemo(() => dataStreamsWithoutAlarms.some(dataStreamHasError), [dataStreamsWithoutAlarms]);
-
-  const isLoading = useMemo(
-    () => !hasError && dataStreamsWithoutAlarms.some(dataStreamIsLoading),
-    [hasError, dataStreamsWithoutAlarms]
-  );
-
-  const visibleData = dataStreamsWithoutAlarms.flatMap(({ data }) => getVisibleData(data, utilizedViewport, false));
+  // dependencies
+  const dataString = JSON.stringify(dataStreams);
+  const thresholdsString = JSON.stringify(thresholds);
+  const utilizedString = JSON.stringify(utilizedViewport);
```

Comment:
I can imagine why this is done, but will everyone recognize it?

Reply:
let me add some comments